hydro_lang/src/viz/url_generation_test.rs

```diff
@@ -0,0 +1,80 @@
+//! Tests for URL generation and compression logic
+
+#[cfg(test)]
+mod tests {
+    use crate::viz::config::VisualizerConfig;
+
+    #[test]
+    fn test_visualizer_config_default() {
+        let config = VisualizerConfig::default();
+        assert!(config.base_url.contains("hydro.run") || config.base_url.contains("localhost"));
+        assert!(config.enable_compression);
+        assert_eq!(config.max_url_length, 4000);
+        assert_eq!(config.min_compression_size, 1000);
+    }
+
+    #[test]
+    fn test_visualizer_config_with_base_url() {
+        let config = VisualizerConfig::with_base_url("https://example.com/viz");
+        assert_eq!(config.base_url, "https://example.com/viz");
+        assert!(config.enable_compression);
+    }
+
+    #[test]
+    fn test_visualizer_config_local() {
+        let config = VisualizerConfig::local();
+        assert_eq!(config.base_url, "http://localhost:3000/hydroscope");
+    }
+
+    #[test]
+    fn test_visualizer_config_without_compression() {
```

Comment:
This feels like a pretty silly test. Especially if we change it to use https://docs.rs/buildstructor/latest/buildstructor/